src/core.jl

```diff
@@ -160,9 +160,12 @@ function DArray(refs)
             for j in 1:(idx_in_dim-1)
                 prevsubidx = ntuple(y -> y == x ? j : subidx[y], length(subidx))
                 prevsize = nsizes[prevsubidx...]
-                startidx += prevsize[x]
+                # presize tuple is empty if reference is not to an array, e.g. a scalar
```

Comment:
The implicit assumption everywhere is that localparts are arrays. I don't know if breaking this can lead to any breakages elsewhere - needs a run through the codebase to verify the same.

Also, if we are going to support this, it should be documented and appropriate comments added. The `init` function in the regular constructor should be able to return a scalar too.